### ISSUE TITLE
NotImplemented The action 'RemotStartTransaction' you are looking for is not found

### ISSUE THREAD

**Issue by yuhui6688** [steve-community/steve#1323]
#### Checklist

- [ ] I checked other issues already and found no answer
- [ ] I checked the documentation and found no answer
- [ ] I am running the latest version and the feature i am requesting is not implemented there

#### Describe the problem you are trying to solve

I have successfully deployed the Steve server and attempted to connect using websocket, but received an error message when sending the Remote Start Transaction. I am not sure why, please advise. Thank you very much!

ERROR Code:
 [4,"4","NotImplemented","The action 'RemotStartTransaction' you are looking for is not found",{}]
...

#### Describe the solution you'd like

...

#### Describe alternative solutions or features you've considered 

...

#### Additional context

...

**Comment by yuhui6688** [comment]
This is a partial sending code：
![image](https://github.com/steve-community/steve/assets/134028855/f8e1b4b1-668c-455d-9929-03d8e599d466)

**Comment by juherr** [comment]
The remote start message is from CSMS to the station and not from the station to the CSMS.

What are you trying to do?

**Comment by juherr** [comment]
What is the simulated App you refer?

I think you want to make a StartTransaction:
![image](https://github.com/steve-community/steve/assets/254697/26fc0bd6-073f-4a42-976d-848c7a53b383)

Have a look at #1000 for the API subject.

**Comment by yuhui6688** [comment]
As shown in the picture below, this is the goal I want to achieve. I don’t know how to do it. Please give me your advice. Thank you very much!
![1701846021437](https://github.com/steve-community/steve/assets/134028855/19dcc027-adb0-400e-8b30-d21cb0c4e66a)

**Comment by yuhui6688** [comment]
Assuming that the ChangePoint status is online, should the App send instructions directly to the Central System or ChangePoint?

**Comment by goekay** [comment]
> should the App send instructions directly to the Central System or ChangePoint?

to the central system, i.e. steve. steve is responsible for all bidirectional communication with charge points.

**Comment by goekay** [comment]
> Does your explanation mean that the APP sends the StartTransaction command to Steve, and then Steve sends the command to the ChargePoint to start the charging function?

yes. but, the api call  "app -> steve" will be an independent one. it is not the StartTransaction or RemoteStartTransaction of ocpp. you can define it as you want, including the payload. steve would need to understand it, map it to a RemoteStartTransaction and send this to the station.

> How does the app know whether it is successful or not? 

the api call "app -> steve" will be waiting for the response from station to arrive at steve. steve will map it to a response payload that you specify and return it. 

something like this:

![remote-start](https://github.com/steve-community/steve/assets/3861031/d400f970-541c-42ce-b103-8b4a532dd7ca)

**Comment by yuhui6688** [comment]
Does Steve have this API, or do you need to implement the service yourself? Thank you so much！

**Comment by goekay** [comment]
it does not, and hence the issue reference above from @juherr

**Comment by yuhui6688** [comment]
@jpires If Steve does not have a relevant API, how can I implement this service interface based on Steve's source code? Thank you so much!

**Comment by yuhui6688** [comment]
@juherr Your reply has given me great encouragement and help! Thank you very much!
I have modified Steve's code according to your method and compiled it successfully, but when I browsed in chrome, I got an error:
![image](https://github.com/steve-community/steve/assets/134028855/4132d673-2328-4bf1-b5d4-8c41be8fed27)
Check the views folder that is not generated at all in the project.
![image](https://github.com/steve-community/steve/assets/134028855/255e7575-d50c-4511-8555-cbae86782796)
 I think I am in trouble. I have tried many methods to no avail. Can you help me?

**Comment by yuhui6688** [comment]
OK,Thanks very much!

**Comment by fnkbsi** [comment]
The compiled jsp-Files can found under "\target\classes\org\apache\jsp\WEB_002dINF\views". As I see in the picture, your folder org is missing. Have you tried a clean recompiling (mvn clean package)?

**Comment by Shivk14** [comment]
@yuhui6688 did u implemented remotestart api and remote stop api?
